File: 2971/FindPolygonWithTheLargestPerimeter.ts
/*
https://leetcode.com/problems/find-polygon-with-the-largest-perimeter/description/?envType=daily-question&envId=2024-02-15
*/

function largestPerimeter(nums: number[]): number {
    nums.sort((A, B)=>(A-B));
    let sz:number=nums.length;
    for(let i:number=sz-1; i>=2; --i){
        let curval:number=nums[i], cursm:number=0;
        for(let j:number=0; j<i; ++j){
            cursm+=nums[j];
        }
        if(cursm>curval) return cursm+curval;
    }
    return -1;
};

File: 2971/FindPolygonWithTheLargestPerimeter.java
/*
https://leetcode.com/problems/find-polygon-with-the-largest-perimeter/description/?envType=daily-question&envId=2024-02-15
*/

class Solution {
    public long largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        int sz=nums.length;
        for(int i=sz-1; i>=0; --i){
            long curval=nums[i], cursm=0, count=0;
            for(int j=0; j<i; ++j){
                cursm+=nums[j];
                count++;
            }
            if(cursm>curval&&count>=2) return cursm+curval;
        }
        return -1;
    }
}